### PR TITLE
Table visits

### PR DESCRIPTION
**Pull request by kamranasif506**
In this PR

- Vets table created
- A JOIN table `specialization` for vets and species created
- A JOIN table `visits` for vets and animals created
- Data inserted in visits and specialization table
  - Who was the last animal seen by William Tatcher?
Screenshot 2023-08-16 124850

  - How many different animals did Stephanie Mendez see?
Screenshot 2023-08-16 125106

  - List all vets and their specialties, including vets with no specialties.
Screenshot 2023-08-16 125943

  - List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
Screenshot 2023-08-16 130214

  - What animal has the most visits to vets?
Screenshot 2023-08-16 130410

  - Who was Maisy Smith's first visit?
Screenshot 2023-08-16 130610

  - Details for most recent visit: animal information, vet information, and date of visit.
Screenshot 2023-08-16 130827

  - How many visits were with a vet that did not specialize in that animal's species?
Screenshot 2023-08-16 131507

  - What specialty should Maisy Smith consider getting? Look for the species she gets the most.
Screenshot 2023-08-16 132030

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -34,4 +34,24 @@ ALTER TABLE animals
...
+    ID INT PRIMARY KEY GENERATED BY DEFAULT AS IDENTITY,
+    vet_id INT NOT NULL,
+    species_id INT NOT NULL
+);
+
+CREATE TABLE visits (
+    ID INT PRIMARY KEY GENERATED BY DEFAULT AS IDENTITY,
+    vet_id INT NOT NULL,
+    animal_id INT NOT NULL
+);
```
Comment (bobb-Rob) [inline schema.sql]: - [ ] Based on the project requirement, this VISIT table should keep track of the visit date. So you need to include a column for `date_of_visit` or `visit_date`, however, you want to call it. 

image

File: data.sql [hunk 2]
```diff
@@ -46,3 +46,42 @@ SET owner_id = CASE
...
+        ('Jack Harkness', 38, '2008-06-08');
+
+INSERT INTO specializations (vet_id, species_id)
+VALUES
+   (1, 1),        
+   (3, 1),        
+   (3, 2),
+   (4, 2);
+
+NSERT INTO visits (animal_id, vet_id, date_of_visit)
```
Comment (bobb-Rob) [inline data.sql:64]: - [ ] So you have `date_of_visit` here but the schema file does not create that column. Please fix

**Review by bobb-Rob — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️
Hi @kamranasif506, 

Good job so far! 👏👏👏 
There are some issues that you still need to work on to go to the next project but you are almost there!  😃 
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## Highlights
✅ The table associations are well done. 👏🏾 
✅ The correct workflow is used. 
✅ Very descriptive Pull Request(PR)
✅ All query responses are as expected

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Gambit142 — APPROVED** [review]
Hi @kamranasif506 👋,

## Status: Approved 🟢   

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

### Highlights
- Everything that creates or changes the structure of the table was saved in the _**schema.sql**_ file ✔️
- Everything that populates tables with data was saved in the _**data.sql**_ file ✔️ 
- Everything that is a query that helped to analyze data was saved in the _**queries.sql**_ file ✔️

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**